packages/assets-controllers/src/TokensController.ts

```diff
@@ -675,7 +715,21 @@ export class TokensController extends BaseController<
     const { suggestedAssets } = this.state;
     suggestedAssets.push(suggestedAssetMeta);
     this.update({ suggestedAssets: [...suggestedAssets] });
-    this.hub.emit('pendingSuggestedAsset', suggestedAssetMeta);
+
+    this.messagingSystem
```

Comment:
Is it worth encapsulating the use of the `ApprovalController` into some private methods to break up these larger methods, so maybe a `_requestApproval`, `_acceptApproval`, and `_rejectApproval`?